### DISCUSSION TITLE
How to switch the tenant in the extension?

### DISCUSSION THREAD

**Discussion by StromFLIX** [microsoft/SynapseVSCode#4] [Q&A]
Always if i sign in it automatically logs me into one tenant, but i am not able to switch tenants somehow. Is there a way to archive this?

**Comment by StromFLIX** [comment]
I think the main problem is in lin 199 / 165 in `src/auth/auth.ts`
```ts
await this.context.globalState.update(CONST.AUTH.TENANT_ID, authResult.account?.homeAccountId);
```

Here it should be changed to `tenantId` probably to make it work, it needs to change everywhere where used:
```ts
const accounts = tenantId ? tenantId.split('.') : [];
  
tenant_id: accounts.length == 2 ? accounts[1] : '',
```

Because it is now not anymore a composite of the userid and the tenantid

A new setting for the tenant ID needs to be added to the extension's configuration. This needs to update the CONST.AUTH.AUTHORITY constant.

**Comment by qixwang** [comment]
@NemesisFLX thanks for the feedback. we will consider adding the support of "Switch tenant" in the future update. could you share how ofter you would need to do the switch?
